File: Travel Package Management Java/src/com/example/travel/DestinationTest.java
package com.example.travel;

import org.junit.Test;
import static org.junit.Assert.*;

public class DestinationTest {
    
    @Test
    public void testAddActivity() {
        Destination paris = new Destination("Paris");
        Activity sightseeing = new Activity("Sightseeing", "Visit landmarks", 100, 15);

        assertTrue(paris.addActivity(sightseeing));
        assertTrue(sightseeing.isAssignedToDestination());
        assertFalse(paris.addActivity(sightseeing)); // Activity already assigned, should return false
    }

    @Test
    public void testAddSameActivityToDifferentDestinations() {
        Destination bangalore = new Destination("Bangalore");
        Destination indore = new Destination("Indore");
        Activity sightseeing = new Activity("Sightseeing", "Visit landmarks", 100, 15);

        assertTrue(bangalore.addActivity(sightseeing));
        assertFalse(indore.addActivity(sightseeing)); // Activity can't be assigned to two destinations
    }
}

File: Travel Package management C++/DestinationTest.cpp
#include "Destination.h"
#include "Activity.h"
#include <gtest/gtest.h>
#include <memory>

TEST(DestinationTest, AddActivity) {
    auto destination = std::make_shared<Destination>("Paris");
    auto activity = std::make_shared<Activity>("Sightseeing", "Visit landmarks", 100, 15);

    EXPECT_TRUE(destination->addActivity(activity));

    EXPECT_TRUE(activity->isAssignedToDestination());

    EXPECT_FALSE(destination->addActivity(activity));
}

TEST(DestinationTest, AddSameActivityToDifferentDestinations) {
    auto destination1 = std::make_shared<Destination>("Bangalore");
    auto destination2 = std::make_shared<Destination>("Indore");
    auto activity = std::make_shared<Activity>("Sightseeing", "Visit landmarks", 100, 15);

    EXPECT_TRUE(destination1->addActivity(activity));
    
    EXPECT_FALSE(destination2->addActivity(activity));

   
    EXPECT_TRUE(activity->isAssignedToDestination());

    EXPECT_FALSE(destination1->addActivity(activity));
}